Orchestrator.py

```diff
@@ -438,4 +390,69 @@ def _computeCellNeighbours(self, cx, cy):
             (cx-cellSize, cy+cellSize), (cx+cellSize, cy-cellSize)
         ]
 
+        return returnVal
+
+    def _computeCellCorners(self, x, y):
+
+        (cxmin, cymin)    = self._xy2cell(x, y)
+        (cxmax, cymax)    = (cxmin + self.MINFEATURESIZE/2, cymin + self.MINFEATURESIZE/2)
+
+        topLeftCorner     = (cxmin, cymin)
+        topRightCorner    = (cxmax, cymin)
+        bottomLeftCorner  = (cxmin, cymax)
+        bottomRightCorner = (cxmax, cymax)
+
+        return [topLeftCorner, topRightCorner, bottomLeftCorner, bottomRightCorner]
+
+    def _checkForCornerFrontier(self, frontier):
+
+        returnVal     = False
+        firstObstacle = None
+
+        # check for obstacle in 1-hop neighbourhood of frontier
+        for n in self._computeCellNeighbours(*frontier):
+            if n in self.cellsObstacle:
+                firstObstacle = n
+                break
+
+        if firstObstacle:
+
+            (cx, cy) = firstObstacle
+
+            # check for second obstacle diagonal to the first one
+            for (nx, ny) in self._computeCellNeighbours(cx, cy):
+
+                if (nx, ny) not in self.cellsObstacle or nx == cx or ny == cy:
+                    # cell not diagonal obstacle to first obstacle detected
+                    # skip this cell
+                    continue
+
+                # if reached here, 2 diagonal obstacles have been detected
+                # find corner where obstacles connect diagonally (common corner)
+                firstObstacleCorners  = set(self._computeCellCorners(cx, cy))
+                secondObstacleCorners = set(self._computeCellCorners(nx, ny))
+
+                commonCorner = firstObstacleCorners.intersection(secondObstacleCorners)
+                (ccx, ccy) = list(commonCorner)[0]
+
+                cellSize       = self.MINFEATURESIZE / 2
+                connectedCells = [
+                    (ccx - cellSize, ccy - cellSize),
+                    (ccx, ccy - cellSize),
+                    (ccx, ccy),
+                    (ccx - cellSize, ccy)
+                ]
+
+                connectedExploredCell = None
+
+                # check that there is one explored cell connected to diagonal obstacles and frontier
+                # at same connecting corner
+                for cell in connectedCells:
+                    if cell in self.cellsExplored:
+                        # remove cell from frontiers if it is already set as one
+                        if frontier in self.cellsFrontier:
+                            self.cellsFrontier.remove(frontier)
+                        returnVal = True
+                        break
+
         return returnVal
```

Comment:
add empty newline (please do from now on, it's the third time I have this comment)